src/bitmap/store.rs

```diff
@@ -136,6 +136,28 @@ impl Store {
         }
     }
 
+    ///
+    /// Pushes `index` in the store only if it is greater than the current maximum value.
+    /// It is up to the caller to have validated value > self.max()
+    ///
+    /// # Panics
+    ///
+    /// If debug_assertions enabled and value is > self.max()
+    pub(crate) fn push_unchecked(&mut self, index: u16) {
+        if cfg!(debug_assertions) {
+            if let Some(max) = self.max() {
+                assert!(index > max, "container max >= index")
+            }
+        }
+        match self {
+            Array(vec) => vec.push(index),
+            Bitmap(bits) => {
+                let (key, bit) = (key(index), bit(index));
+                bits[key] |= 1 << bit;
+            }
+        }
+    }
+
```

Comment:
```suggestion
        if cfg!(debug_assertions) {
            assert!(self.push(index), "container max >= index")
        } else {
            match self {
                Array(vec) => vec.push(index),
                Bitmap(bits) => {
                    let (key, bit) = (key(index), bit(index));
                    bits[key] |= 1 << bit;
                }
            }
        }
    }

```

Reply:
Cant do this with your other suggestion of having push call `push_unchecked`

Reply:
ahah yeah, obviously, sorry didn't though about both at the same time 😄